### PR TITLE
cxx-qt-lib: ensure that QVariant{Hash,List,Map} are registered for QML

### LINKED ISSUE #533: QVariantMap
The cxx-qt-lib appears to not have a QVariantMap. However, it does have a QMap-QString-QVariant. Which I can use, but since it is not recognized as a QVariantMap in QML, it must first pass through C++ before QML can pick it up. I've worked around this with a C++ bridge that passes the params to rust functions and the returned QMap-QString-QVariant back to QML as a QVariantMap and QT seems to do the conversion automatically since it's just an alias for the same thing.

Any idea why Rust can't send those back to QML without first going through C++?

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Closes #533

### CODE REVIEW

File: crates/cxx-qt-lib/src/std_types.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+#include <cstdint>
+
+namespace rust {
+namespace cxxqtlib1 {
+
+// Ensure that std int types are registered
+// so that they can be used with QML in Qt 5
+
+static const int register_i8 =
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/std_types.cpp:22]: this causes the extension plugin tests to fail.

Is this the same problem as the `+whole-archive` option where things are compiled out as they aren't used ?

Eg this `std_types.cpp` doesn't get executed ...

Comment (ahayzen-kdab) [reply crates/cxx-qt-lib/src/std_types.cpp:22]: Moving the extension plugin example to use WHOLE_ARCHIVE in the CMake like other examples fixes this...

**Review by LeonMatthesKDAB — COMMENTED** [review]
We will also want to update the book section [Using a Custom Type with Containers or QVariant](https://kdab.github.io/cxx-qt/book/concepts/types.html#using-a-custom-type-with-containers-or-qvariant) to explain that you will have to call qmlRegisterType for your own types as well.

**Comment by ahayzen-kdab** [comment]
> We will also want to update the book section [Using a Custom Type with Containers or QVariant](https://kdab.github.io/cxx-qt/book/concepts/types.html#using-a-custom-type-with-containers-or-qvariant) to explain that you will have to call qmlRegisterType for your own types as well.

Done

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**